cmd/server/main.go

```diff
@@ -155,11 +161,23 @@ func buildCLI() *cli.App {
 					logger.Info("Dynamic config client is not configured. Using noop client.")
 				}
 
-				authorizer, err := authorization.GetAuthorizerFromConfig(
-					&cfg.Global.Authorization,
-				)
-				if err != nil {
-					return cli.Exit(fmt.Sprintf("Unable to instantiate authorizer. Error: %v", err), 1)
+				var authorizer authorization.Authorizer
+				if no_auth {
+					authorizer = authorization.NewNoopAuthorizer()
+				} else {
+					authorizer, err = authorization.GetAuthorizerFromConfig(
+						&cfg.Global.Authorization,
+					)
+					if err != nil {
+						return cli.Exit(fmt.Sprintf("Unable to instantiate authorizer. Error: %v", err), 1)
+					}
+					if authorization.IsNoopAuthorizer(authorizer) {
```

Comment:
remove the if no_auth check at line 165, and add it here:
if authorization.IsNoopAuthorizer(authorizer) && !no_auth {
  // using noopAuthorizer without --no-auth flag
  // log warning now, and fetal in future release.
}

Reply:
And probably change the flag to --allow-no-auth because in case this flag is present and config specified a valid authorizer, we would still use that configured authorizer.